app/Http/Controllers/HeadOfProgramController.php

```diff
@@ -33,36 +34,57 @@ public function actives(): Response
                 unset($program->latestEpisode);
                 return $program;
             })->onEachSide(5);
-        dd(json_encode($programs, JSON_PRETTY_PRINT));
-        return Inertia::render('CHANGEME', $programs);
+        return Inertia::render('HeadOfProgram/ActiveProgram', [
+            'programs' => $programs
+        ]);
     }
 
-    public function program(Program $program): Response
+    public function draftProgram(Program $program): Response
     {
-        // TODO: status property
-        $program->episodes = $program->episodes();
-        dd(json_encode($program));
-        return Inertia::render('CHANGEME', $program);
+        if ($program->episodes()->exists()) {
+            abort(404);
+        }
+
+        return Inertia::render('HeadOfProgram/ProgramDetail', [
+            'program' => $program,
+            'source' => 'drafts'
+        ]);
+    }
+
+    public function activeProgram(Program $program): Response
+    {
+        if (!$program->episodes()->exists()) {
+            abort(404);
+        }
+
+        $program->load('episodes');
+
+        return Inertia::render('HeadOfProgram/ProgramDetail', [
+            'program' => $program,
+            'source' => 'actives'
+        ]);
     }
 
     public function create(CreateProgramRequest $req): HttpResponse|ResponseFactory
     {
         try {
             $payload = $req->validated();
             Program::create($payload);
-        } catch (Exception) {
+        } catch (Exception $e) {
+            dd($e->getMessage());
             return response(status: 500);
         }
         return response(status: 201);
     }
 
-    public function update(UpdateProgramRequest $req): HttpResponse|ResponseFactory
+    public function update(UpdateProgramRequest $req, $slug): HttpResponse|ResponseFactory
     {
         try {
+            $program = Program::where('slug', $slug)->firstOrFail();
```

Comment:
it can be much simpler than this by binding the slug url param straight to a program model, i'll help you with the fix changes

Reply:
fixed in 8ab61711b47e678b9caf16d8d500ee61bc0d25e3

Reply:
okey mas aman